### ISSUE TITLE
 UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'records' of null

### ISSUE THREAD

**Issue by Nerzal** [rfennell/AzurePipelines#773]
Hi first of all, thank u for your extensions :) 

**Describe the bug**
I tried to generate release notes using the sample template from the documentation. 
This results in the following error:

```
(node:3680) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'records' of null
    at /home/vsts/work/_tasks/XplatGenerateReleaseNotes_7b0a5887-75c4-4ade-a915-f92a93c934cb/3.2.2/ReleaseNotesFunctions.js:565:74
    at Generator.next (<anonymous>)
    at fulfilled (/home/vsts/work/_tasks/XplatGenerateReleaseNotes_7b0a5887-75c4-4ade-a915-f92a93c934cb/3.2.2/ReleaseNotesFunctions.js:5:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:3680) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:3680) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

*Remember that there are a number of Azure DevOps (VSTS) extensions in this repo so please say which one you are using, and if an extension contains multiple tasks which task also*
Extension: XplatGenerateReleaseNotes

**To Reproduce**
I use the following configuration: 

```
- task: XplatGenerateReleaseNotes@3
  inputs:
    outputfile: '$(Build.ArtifactStagingDirectory)\releasenotes.md'
    outputVariableName: 'kycnow-dev-release'
    templateLocation: 'InLine'
    inlinetemplate: |
      **Release Notes**
      
      [[_TOC_]]  
       
      # Notes for release  {{releaseDetails.releaseDefinition.name}}    
      **Release Number**  : {{releaseDetails.name}}
      **Release completed** : {{releaseDetails.modifiedOn}}     
      **Build Number**: {{buildDetails.id}}
      **Compared Release Number**  : {{compareReleaseDetails.name}}    
      **Build Trigger PR Number**: {{lookup buildDetails.triggerInfo 'pr.number'}} 

      # Associated Pull Requests ({{pullRequests.length}})
      {{#forEach pullRequests}}
      {{#if isFirst}}### Associated Pull Requests (only shown if  PR) {{/if}}
      *  **PR {{this.id}}**  {{this.title}}
      {{/forEach}}

      # Builds with associated WI/CS/Tests ({{builds.length}})
      {{#forEach builds}}
      {{#if isFirst}}## Builds {{/if}}
      ##  Build {{this.build.buildNumber}}
      {{#forEach this.commits}}
      {{#if isFirst}}### Commits {{/if}}
      - CS {{this.id}}
      {{/forEach}}
      {{#forEach this.workitems}}
      {{#if isFirst}}### Workitems {{/if}}
      - WI {{this.id}}
      {{/forEach}} 
      {{#forEach this.tests}}
      {{#if isFirst}}### Tests {{/if}}
      - Test {{this.id}} 
        -  Name: {{this.testCase.name}}
        -  Outcome: {{this.outcome}}
      {{/forEach}} 
      {{/forEach}}

      # Global list of WI ({{workItems.length}})
      {{#forEach this.workItems}}
      {{#if isFirst}}### WorkItems {{/if}}
      *  **{{this.id}}**  {{lookup this.fields 'System.Title'}}
        - **WIT** {{lookup this.fields 'System.WorkItemType'}} 
        - **Tags** {{lookup this.fields 'System.Tags'}}
        - **Assigned** {{#with (lookup this.fields 'System.AssignedTo')}} {{displayName}} {{/with}}
        - **Description** {{{lookup this.fields 'System.Description'}}}
        - **Parents**
      {{#forEach this.relations}}
      {{#if (contains this.attributes.name 'Parent')}}
      {{#with (lookup_a_work_item ../../relatedWorkItems  this.url)}}
            - {{this.id}} - {{lookup this.fields 'System.Title'}} 
      {{/with}}
      {{/if}}
      {{/forEach}} 
        - **Children**
      {{#forEach this.relations}}
      {{#if (contains this.attributes.name 'Child')}}
      {{#with (lookup_a_work_item ../../relatedWorkItems  this.url)}}
            - {{this.id}} - {{lookup this.fields 'System.Title'}} 
      {{/with}}
      {{/if}}
      {{/forEach}} 
      {{/forEach}} 

      # Global list of CS ({{commits.length}})
      {{#forEach commits}}
      {{#if isFirst}}### Associated commits{{/if}}
      * ** ID{{this.id}}** 
        -  **Message:** {{this.message}}
        -  **Commited by:** {{this.author.displayName}} 
        -  **FileCount:** {{this.changes.length}} 
      {{#forEach this.changes}}
            -  **File path (TFVC or TfsGit):** {{this.item.path}}  
            -  **File filename (GitHub):** {{this.filename}}  
      {{/forEach}}
      {{/forEach}}
    checkStage: true
    overrideStageName: 'dev'
    stopOnRedeploy: false
    sortWi: true
    dumpPayloadToConsole: false
    dumpPayloadToFile: false
    replaceFile: true
    getParentsAndChildren: true
```

**Expected behavior**
No error is being thrown and release notes get generated.

**Screenshots**

**Logs**

```
Starting: XplatGenerateReleaseNotes
==============================================================================
Task         : Generate Release Notes (Crossplatform)
Description  : Generates a release notes file in a format of your choice from the build or release history
Version      : 3.2.2
Author       : Black Marble
Help         : Version: 3.2.2. [More Information](https://github.com/rfennell/AzurePipelines/wiki/GenerateReleaseNotes---Node-based-Cross-Platform-Task/)
==============================================================================
Getting the current build details
Overriding current stage '__default' with 'dev'
Getting items associated the builds since the last successful build to the stage 'dev'
Comparing 441 against 441
Ignore compare against self
Comparing 440 against 441
Comparing 439 against 441
Comparing 438 against 441
Comparing 437 against 441
Comparing 436 against 441
(node:3680) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'records' of null
    at /home/vsts/work/_tasks/XplatGenerateReleaseNotes_7b0a5887-75c4-4ade-a915-f92a93c934cb/3.2.2/ReleaseNotesFunctions.js:565:74
    at Generator.next (<anonymous>)
    at fulfilled (/home/vsts/work/_tasks/XplatGenerateReleaseNotes_7b0a5887-75c4-4ade-a915-f92a93c934cb/3.2.2/ReleaseNotesFunctions.js:5:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:3680) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:3680) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
Finishing: XplatGenerateReleaseNotes
```

**What is the issue see:**
 - Azure DevOps Services (VSTS)

**Additional context**

**Comment by Nerzal** [comment]
The Problem does not occur, when setting 

```
   checkStage: false
```

**Comment by rfennell** [comment]
The issue appears to be that when looking for the last successful build it find a build but cannot get any timeline details. As I had assumed there is always a timeline it fails. My guess is that it is a cancelled build

I will add a null check to trap this and hopefully get a new release today

**Comment by rfennell** [comment]
and yes ` checkStage: false` is a different code path, so that makes sense

Thank @Nerzal for the detailed fault report, it really helps

**Comment by rfennell** [comment]
This is being released as 3.3.10

Let me know how it goes, re-open if you still have an issue

**Comment by Nerzal** [comment]
It seems to work, the log now shows for the problematic builds. 
Thank you for the fast response and fix <3 

> Skipping check as no timeline available for this build
